packages/nouns-contracts/contracts/governance/data/NounsDAOData.sol

```diff
@@ -0,0 +1,272 @@
+// SPDX-License-Identifier: GPL-3.0
+
+/// @title Nouns DAO Data Contract
+
+/*********************************
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ * ░░░░░░█████████░░█████████░░░ *
+ * ░░░░░░██░░░████░░██░░░████░░░ *
+ * ░░██████░░░████████░░░████░░░ *
+ * ░░██░░██░░░████░░██░░░████░░░ *
+ * ░░██░░██░░░████░░██░░░████░░░ *
+ * ░░░░░░█████████░░█████████░░░ *
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ *********************************/
+
+pragma solidity ^0.8.6;
+
+import { OwnableUpgradeable } from '@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol';
+import { NounsDAOV3Proposals } from '../NounsDAOV3Proposals.sol';
+import { NounsTokenLike } from '../NounsDAOInterfaces.sol';
+
+contract NounsDAOData is OwnableUpgradeable {
+    /**
+     * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
+     *   ERRORS
+     * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
+     */
+
+    error MustBeNounerOrPaySufficientFee();
+    error SlugAlreadyUsed();
+    error SlugDoesNotExist();
+    error MustBeNouner();
+
+    /**
+     * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
+     *   EVENTS
+     * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
+     */
+
+    event ProposalCandidateCreated(
+        address indexed msgSender,
+        address[] targets,
+        uint256[] values,
+        string[] signatures,
+        bytes[] calldatas,
+        string description,
+        string slug,
+        bytes32 encodedProposalHash
+    );
+    event ProposalCandidateUpdated(
+        address indexed msgSender,
+        address[] targets,
+        uint256[] values,
+        string[] signatures,
+        bytes[] calldatas,
+        string description,
+        string slug,
+        bytes32 encodedProposalHash
+    );
+    event ProposalCandidateCanceled(address indexed msgSender, string slug);
+    event SignatureAdded(
+        address indexed msgSender,
+        address indexed signer,
+        bytes sig,
+        uint256 expirationTimestamp,
+        bytes32 encodedPropHash,
+        string reason
+    );
+    event FeedbackSent(address indexed msgSender, uint256 proposalId, uint8 support, string reason);
+    event CreateCandidateCostSet(uint256 oldCreateCandidateCost, uint256 newCreateCandidateCost);
+    event UpdateCandidateCostSet(uint256 oldUpdateCandidateCost, uint256 newUpdateCandidateCost);
+
+    /**
+     * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
+     *   STATE
+     * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
+     */
+
+    /// @notice The cost non-Nouners must pay in ETH in order to emit a new proposal candidate event from this contract.
+    uint256 public createCandidateCost;
+    /// @notice The cost non-Nouners must pay in ETH in order to emit a proposal candidate update event from this contract.
+    uint256 public updateCandidateCost;
+    /// @notice The Nouns token contract.
+    NounsTokenLike public nounsToken;
```

Comment:
@eladmallel this can probably be immutable?
would need to be set in constructor instead of initializer then

Reply:
agreed, will do

Reply:
done